### DISCUSSION TITLE
My command is not running, please help me out

### DISCUSSION THREAD

**Discussion by TechPro424** [FabricMC/fabric-api#2847] [Mod Dev Support] [answered]
My command doesn't execute the `run` function at all, and instead errors out.
I know it does not execute the `run` function because I added a `Logger.info("hi")` to the top of the run function but it doesnt log
The logger logs properly in other places in the mod though so its not a problem with the logger
Check the gist for the code and error
https://gist.github.com/TechPro424/86424580c8e2ac09e779aaf7b0f96fb0

**Comment by Juuxel** [comment, marked as answer]
The `executes` must be on the last command node in the chain, so
```java
        dispatcher.register(CommandManager.literal("addIdToBlacklist")
        .then(RequiredArgumentBuilder.argument("id", StringArgumentType.greedyString())
            .executes(AddIdToBlacklistCommand::run)));
```
(If it's hard to spot, I moved the `executes` inside the `then`)

**Reply by TechPro424** [reply]
![image](https://user-images.githubusercontent.com/86620085/212470312-68784e9f-2114-410d-922b-7f2f55f72a05.png)
It errors out
The fix to this would be to change the type of the `context` parameter in the `run` function from `CommandContext<ServerCommandSource>` to `CommandContext<Object>`

But that errors out and does not let me use `sendFeedback()` which I need to use, and also throws an error in the `then()`
![image](https://user-images.githubusercontent.com/86620085/212470441-607f1418-8a3d-448a-8808-72e121bc2adf.png)

**Reply by Juuxel** [reply]
You can use `CommandManager.argument` to avoid the errors with generics (just like `CommandManager.literal`)
